controller/registry/dockerclient.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+"""Support the Deis workflow by manipulating and publishing Docker images."""
+
+from __future__ import unicode_literals
+import io
+import logging
+
+from django.conf import settings
+from rest_framework.exceptions import PermissionDenied
+from simpleflock import SimpleFlock
+import docker
+
+logger = logging.getLogger(__name__)
+
+
+class DockerClient(object):
+    """Use the Docker API to pull, tag, build, and push images to deis-registry."""
+
+    FLOCKFILE = '/tmp/controller-pull'
+
+    def __init__(self):
+        self.client = docker.Client(version='auto')
+        self.registry = settings.REGISTRY_HOST + ':' + str(settings.REGISTRY_PORT)
+
+    def publish_release(self, source, config, target, deis_registry):
+        """Update a source Docker image with environment config and publish it to deis-registry."""
+        # get the source repository name and tag
+        src_name, src_tag = docker.utils.parse_repository_tag(source)
+        # get the target repository name and tag
+        name, tag = docker.utils.parse_repository_tag(target)
+        # strip any "http://host.domain:port" prefix from the target repository name,
+        # since we always publish to the Deis registry
+        paths = name.split('/')
+        name = '/'.join(p for p in paths if p and '.' not in p and ':' not in p)
```

Comment:
are there tests covering this "stripping"? Might be useful so we can cover edge cases in the future if necessary :)